src/core/lib/iomgr/unix_sockets_posix.cc

```diff
@@ -110,8 +110,7 @@ std::string grpc_sockaddr_to_uri_unix_if_possible(
   }
   if (((struct sockaddr_un*)addr)->sun_path[0] == '\0' &&
       ((struct sockaddr_un*)addr)->sun_path[1] != '\0') {
-    // If this is an abstract unix socket, return it with a '@' in the name.
-    return absl::StrCat("unix:@", ((struct sockaddr_un*)addr)->sun_path + 1);
+    return absl::StrCat("unix-abstract:", ((struct sockaddr_un*)addr)->sun_path + 1);
```

Comment:
This is not going to work in the presence of a NUL byte within the name.  Instead, I think we need to do something like this:

```
struct sockaddr_un* un =
      reinterpret_cast<struct sockaddr_un*>(resolved_addr->addr);
return absl::StrCat(
    "unix-abstract:",
    absl::string_view(un->sun_path + 1, resolved_addr->len - sizeof(un->sun_family) - 1));
```

Reply:
Please make this change anyway, since it will work fine even if there's no NUL byte, but we won't have to change it later when we fix the URI parser.

Reply:
Done